packages/editor/src/hooks/useBrowserCheck.ts

```diff
@@ -0,0 +1,22 @@
+import { useEffect, useState } from 'react'
+
+export const useBrowserCheck = () => {
+  const [unsupported, setUnsupported] = useState(false)
+
+  useEffect(() => {
+    const detectBrowser = () => {
+      const userAgent = navigator.userAgent
+
+      const isChrome = /Chrome/.test(userAgent) && /Google Inc/.test(navigator.vendor)
+      const isSafari = /^((?!chrome|android).)*safari/i.test(userAgent) && /Apple Computer/.test(navigator.vendor)
+
+      if (!isChrome && !isSafari) {
+        setUnsupported(true)
+      }
+    }
+
+    detectBrowser()
+  }, [])
+
+  return unsupported
+}
```

Comment:
this can be transformed into a normal function instead of a hook
because this returns the output only when the component mounts which is the same as a normal return from a function